### PR TITLE
Gram Vaani importer

### PR DESCRIPTION
**Pull request by kdavis-mozilla**

### CODE REVIEW

File: bin/import_gram_vaani.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        "--very-verbose",
+        action="store_const",
+        required=False,
+        help="set loglevel to DEBUG",
+        dest="loglevel",
+        const=logging.DEBUG,
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: "file" seems very generic. How about something more descriptive like "source_csv"?
I'd prefer a positional argument, as it feels more natural for a required setting.

File: bin/import_gram_vaani.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        required=True,
+        help="Path to the GramVaani csv",
+        dest="csv_filename",
+    )
+    parser.add_argument(
+        "-d",
+        "--directory",
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: "directory" seems very generic. How about something more descriptive like "target_dir"?
I'd prefer a positional argument, as it feels more natural for a required setting.

File: bin/import_gram_vaani.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+        required=True,
+        help="Directory in which to save the importer GramVaani data",
+        dest="directory",
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: Is this required?

File: bin/import_gram_vaani.py [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+        )
+        data.dropna(inplace=True)
+        _logger.info("Parsed %d lines csv file." % len(data))
+        return data
+
+class GramVaaniDownloader:
+    """GramVaaniDownloader downloads a GramVaani dataset.
+    Args:
+      gram_vaani_csv (GramVaaniCSV): A GramVaaniCSV representing the data to download
+      directory (str): The path to download the data from
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: Shouldn't it  be "The path to download the data **to**"?

File: bin/import_gram_vaani.py [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+    Attributes:
+        data (:class:`pandas.DataFrame`): `pandas.DataFrame` Containing the GramVaani csv data
+    """
+
+    def __init__(self, gram_vaani_csv, directory):
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: How about a more descriptive parameter name for `directory`?

File: bin/import_gram_vaani.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+      directory (str): The path to download the data from
+      mp3_directory (os.path): The path containing the GramVaani mp3's
+    Attributes:
+        directory (str): The target directory passed as a command line argument
+        mp3_directory (os.path): The path containing the GramVaani mp3's
+    """
+
+    def __init__(self, directory, mp3_directory):
+        self.directory = directory
+        self.mp3_directory = Path(mp3_directory)
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: nit: Why not using `Path` class also for the other paths?

File: bin/import_gram_vaani.py [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+        self.valid = pd.DataFrame(columns=["wav_filename","wav_filesize","transcript"])
+        self.train = pd.DataFrame(columns=["wav_filename","wav_filesize","transcript"])
+        self.dev = pd.DataFrame(columns=["wav_filename","wav_filesize","transcript"])
+        self.test = pd.DataFrame(columns=["wav_filename","wav_filesize","transcript"])
+
+    def create(self):
+        self._convert_csv_data_to_raw_data()
+        self.raw.index = range(len(self.raw.index))
+        self.valid = self.raw[self._is_valid_raw_rows()]
+        self.valid = self.valid.sample(frac=1).reset_index(drop=True)
```
Comment (tilmankamp) [inline bin/import_gram_vaani.py]: Took me a bit to figure out that this is just "rolling the dice". :-)

File: bin/import_gram_vaani.py (hunk 1 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: OK

File: bin/import_gram_vaani.py (hunk 2 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: OK, I'll change the name.

File: bin/import_gram_vaani.py (hunk 3 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: As there is currently no fallback, yes.

File: bin/import_gram_vaani.py (hunk 4 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: Yes it should

File: bin/import_gram_vaani.py (hunk 5 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: OK, will change the name.

File: bin/import_gram_vaani.py (hunk 6 above)
Comment (kdavis-mozilla) [reply bin/import_gram_vaani.py]: Looking at the code in retrospect, I think this Path is there as the code...

```python
...
for mp3_filename in self.mp3_directory.glob('**/*.mp3'):
    wav_filename = path.join(wav_directory, os.path.splitext(os.path.basename(mp3_filename))[0] + ".wav")
    if not path.exists(wav_filename):
...
```

would have to change to

```python
...
mp3_directory = Path(self.mp3_directory)
for mp3_filename in mp3_directory.glob('**/*.mp3'):
    wav_filename = path.join(wav_directory, os.path.splitext(os.path.basename(mp3_filename))[0] + ".wav")
    if not path.exists(wav_filename):
...
```

If I was consistent in using strings.

If I used Path everywhere I don't think the usage of self.directory would change.

I'm not sure if there's a big "win" for switching all code to use Path.

**Pull request merged**